scene/main/viewport.cpp

```diff
@@ -3848,7 +3848,67 @@ Viewport *Viewport::get_parent_viewport() const {
 
 void Viewport::set_embedding_subwindows(bool p_embed) {
 	ERR_THREAD_GUARD;
-	gui.embed_subwindows_hint = p_embed;
+	if (gui.embed_subwindows_hint == p_embed) {
+		return;
+	}
+
+	bool allow_change = true;
+
+	if (gui.embed_subwindows_hint) {
+		if (gui.sub_windows.size() > 0) {
+			// Prevent change when this viewport has embedded windows.
+			allow_change = false;
+		}
+	} else {
+		Viewport *vp = this;
+		while (true) {
+			if (!vp->get_parent()) {
+				break;
+			}
+			vp = vp->get_parent()->get_viewport();
+			if (!vp) {
+				// Necessary in cases when function is called on viewports that are not in the scene tree.
+				break;
+			}
```

Comment:
We should be able to check that before the loop. There is no situation when our current viewport is in tree, but its parents somehow aren't.

Reply:
Good idea thanks, done.